### DISCUSSION TITLE
Web3Model in React Native

### DISCUSSION THREAD

**Discussion by sumukus** [WalletConnect/walletconnect-monorepo#2542] [Developer Support]
I am new to it. I am exploring how to use web3modal in React native app.  I followed the steps given on the official website but I got an error and I don't know what to do next. 

My react example of react native app can be found at https://github.com/sumukus/gcitCert

The file where I have written the logic is inside the Home.js file inside the screens folder
      https://github.com/sumukus/gcitCert/blob/main/screens/Home.js

Before I click on Connect button I am getting the error.

Android Bundling complete 117ms
 LOG  Shims Injected:
 LOG    - atob
 LOG    - btoa
 LOG    - nextTick
 LOG    - FileReader.prototype.readAsArrayBuffer
 ERROR  {"context": "core"} {"context": "core/relayer"} [Error: WebSocket connection closed abnormally with code: 3000 (Authorization error: Unauthorized: invalid key)]
 INFO  {"context": "core"} Core Initialization Success
 INFO  {"context": "client"} SignClient Initialization Success
 WARN  Possible Unhandled Promise Rejection (id: 0):
    at apply (native)

Can anyone help me? It would be great.

**Comment by zmanian** [comment]
I am also facing this invalid key error with wagmi 0.12.18

Authorization error: Unauthorized: invalid key

**Comment by ignaciosantise** [comment]
hey @sumukus that error appears when you don't provide a valid `projectId`

You can follow our docs to install the SDK -> https://docs.walletconnect.com/2.0/reactnative/walletconnectmodal/installation#obtain-project-id
